### PR TITLE
Index function to get all peer reviews that have been escalated

### PR DESCRIPTION
**Pull request by mehalshah**

### CODE REVIEW

File: dashboard/app/controllers/api/v1/peer_review_submissions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      escalated_submissions[review.user_level.id] = {
+        submitter: review.submitter.name,
+        course_name: plc_course_unit.plc_course.name,
+        unit_name: plc_course_unit.name,
+        level_name: review.level.name,
+        submission_date: review.created_at,
+        escalated_date: review.updated_at,
+        review_id: review.id
+      }
+    end
```
Comment (aoby) [inline dashboard/app/controllers/api/v1/peer_review_submissions_controller.rb:31]: Should this be handled in a Serializer?
Then we could do something like

```ruby
render json: escalated_reviews, each_serializer: Api::V1::PeerReviewSubmissionSerializer
```

File: dashboard/app/models/plc/course_unit.rb [hunk 2]
```diff
@@ -26,6 +26,11 @@ class Plc::CourseUnit < ActiveRecord::Base
 
   validates :plc_course, presence: true
 
+  # TODO: Migrate unit_name to name for consistency
```
Comment (aoby) [inline dashboard/app/models/plc/course_unit.rb]: Please add your name here so we know who's on the hook :)

File: dashboard/test/controllers/api/v1/peer_review_submissions_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    response = JSON.parse(@response.body)
+    expected_response = [{
+      submitter: @submitter.name,
+      course_name: @course_unit.plc_course.name,
+      unit_name: @course_unit.name,
+      level_name: @level_1.name
+    }.stringify_keys]
+    assert_equal expected_response, response.map {|r| r.reject {|k, _| ['submission_date', 'escalated_date', 'review_id'].include? k}}
+  end
+end
```
Comment (aoby) [inline dashboard/test/controllers/api/v1/peer_review_submissions_controller_test.rb]: Please add basic auth tests: make sure users with the `plc_reviewer` permission can see it and those without cannot.

File: dashboard/app/controllers/api/v1/peer_review_submissions_controller.rb (hunk 1 above)
Comment (mehalshah) [reply dashboard/app/controllers/api/v1/peer_review_submissions_controller.rb:31]: I don't think a serializer is appropriate - we are going from n peer reviews to somewhere between .5n and n submissions. The reason we're doing the hash map is so if a submission has two escalated reviews, it only occurs once it what's returned.

Comment (aoby) [reply dashboard/app/controllers/api/v1/peer_review_submissions_controller.rb:31]: Oh that makes sense. I didn't notice the deduping

**Pull request merged**